src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -1051,14 +996,31 @@ def _visit_reduce(self, node: itir.FunCall):
                 outputs=output_mapping,
                 symbol_mapping=symbol_mapping,
                 input_nodes={arg.value.data: arg.value for arg in args},
-                output_nodes={result_name: result_access},
+                output_nodes={reduce_input_name: reduce_input_node},
             )
 
-            # we apply map fusion only to the nested-SDFG which is generated for the reduction operator
-            # the purpose is to keep the ITIR-visitor program simple and to clean up the generated SDFG
-            self.context.body.apply_transformations_repeated([MapFusion], validate=False)
+        reduce_input_desc = reduce_input_node.desc(self.context.body)
+
+        result_name = unique_var_name()
+        self.context.body.add_array(result_name, (1,), reduce_dtype, transient=True)
```

Comment:
Why is this always a number?

Reply:
As I understand, reduction will return an aggregated value (a scalar) over the iteration space.

Reply:
I can foresee the next question: why we do not use `add_scalar` then? The reason is that the DaCe reduce library node is expecting an array. And the reason for this is that the DaCe library node is generic enough to support reduction along a subset of the input shape dimensions.

Reply:
Actually, I had my fair share of DaCe "Array-Scalar" Inconsistency. Sometimes DaCe allows the use of a scalar but actually does as it where an array, without any warning.